src/main/java/io/beanmapper/BeanMapper.java

```diff
@@ -146,7 +146,7 @@ public <S, T> Collection<T> map(Collection<S> collection, Class<T> elementInColl
      * @param <T> the class type of an element in the target list
      * @return the target list with mapped source list elements
      */
-    public <S, T> List<T> map(List<S> list, Class<T> elementInListClass) {
+    public <S, T> Iterable<T> map(Iterable<S> list, Class<T> elementInListClass) {
```

Comment:
BeanMapper#map(List, Class) is a convenience method which allows for mapping from a List, to a List. In this case, the occurrence of List is a required part of the functionality. As such, this change should be reverted. However, BeanMapper#map(Iterable, Class) might be a good addition. Adding such functionality would probably be best placed in a separate issue and PR.